bin/node/executor/tests/basic.rs

```diff
@@ -697,6 +697,8 @@ fn native_big_block_import_succeeds() {
 fn native_big_block_import_fails_on_fallback() {
 	let mut t = new_test_ext(compact_code_unwrap(), false);
 
+	set_heap_pages(&mut t.ext(), sp_state_machine::HEAP_PAGES_TEST_LEGACY);
```

Comment:
In `TestExternalities::new_with_code` we where always setting heap pages to 8:

https://github.com/paritytech/substrate/blob/6749c701900747accaa35eaf2101586f09baef9e/primitives/state-machine/src/testing.rs#L106

We needed to remove that though so we could override heap_pages when using wasm with `NativeExecutor`.

However a few tests broke, some just because the storage root changed, but others because they implicitly expected an OOM error (like above). Thus, in order to keep the tests passing then I need to explicitly set the heap pages to ensure an OOM error

@basti given this, would you still like me to remove `HEAP_PAGES_TEST_LEGACY` as you mention here https://github.com/paritytech/substrate/pull/8966/files#r647753194 ? And if yes what should I instead do to maintain the tests?

Reply:
Yeah I pointed this out. I know this :D 

However, if the test is required to oom. Just use `set_heap_pages(8)` and add a comment. The legacy stuff can be removed.

Just set the heap pages here and remove this constant. Then we are fine.

And my handle is actually @bkchr :D Sorry at you other Basti